_posts/zh_CN/posts/2016-01-26-segwit-benefits.md

```diff
@@ -0,0 +1,209 @@
+ * [Ethan Heilman 描述发现算法圈](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-January/012202.html)
+ * [Rusty Russell 计算进行攻击的成本](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-January/012227.html)
+ * [Anthony Towns 用周期算法剥削交易](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-January/012218.html)
+ * [Gavin Andresen 总结](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-January/012234.html)
+
+## 脚本版
+
+更改比特币的脚本允许提高安全性和改进功能。然而，脚本的设计只允许向后兼容（软分叉）通过更换其中一个新的操作码的十个额外 OP_NOP 操作码能让脚本失效，
+要不就没有任何可能。这足够让许多变化产生 - 如引入一个新的签名方法或类似 OP_CLTV 功能，但两者无法单独操作（例如，OP_CLTV通常必须由一个 OP_DROP 陪同），
+并不能用于作为连接两个字符串。
+
+Segwit 包括脚本版本号解决此，在非segwit交易也需要用一个额外操作码，通过增加脚本版本支持。
+
+### 谁受惠?
+
+脚本操作码会使比特币高级脚本容易化。这包括改变，如引进的 Schnorr 签名，使用一键恢复收缩签名大小，支持侧链，或创建使用 Merklized 抽象语法
+（MAST） 和其他研究级想法。
+
+## 减少 UTXO 增长
+
+（UTXO）数据库是由每个验证 Bitcoin 节点维持，以确定新的交易是否有效或欺诈。为网络的有效运行，该数据库必须非常快查询和修改，
+并且理想地应当能够适合主存储器（RAM），因此保持该数据库的小是很重要的。
+
+当比特币增长这就变得困难，因为每个新用户必须有自己的至少一个UTXO条目，每个用户有多个条目，保护隐私和灵活性，或提供为后盾，为支付渠道。
+
+Segwit改善了的情况通过签名数据，这不会影响UTXO大小，成本比原本UTXO组大小的数据少75％。预计这将鼓励用户使用，大幅度减少费用，
+并鼓励开发人员设计方式，也将减少对UTXO的影响智能合同和新功能设置UTXO影响。
+
+因为segwit是软分叉变化并不增加基础大小，所述UTXO最坏情况下生长速度就保持不变。
+
+### 谁受惠?
+
+减少 UTXO 增长将有利于矿工，企业，完整的节点使用者，有助于维持在Bitcoin网络的安全随着更多的用户进入系统的用户。
+用户和开发帮助减少UTXO的增长将享有较低的费用，相比忽视交易对UTXO增长的影响使用者。
+
+### 更多资料
+
+ * [Statoshi UTXO 排行榜](http://statoshi.info/dashboard/db/unspent-transaction-output-set)
+ 
+## 提防诈骗
+
+随着比特币用户群的扩大，验证整个链接变得更加昂贵。为了保持比特币的分散，不可信的性质，重要的是让不能对整个链接验证的用户至少能便宜地验证。
+
+Segwit改善了情况并通过允许未来软叉延长证人结构并包括承诺数据，这将允许轻（SPV）客户端来执行一致的规则，例如，如在一并引入比特币的数量，
+并在大量中数目中使用 sigops。
+
+### 谁受惠?
+
+提防欺诈允许SPV用户帮助执行比特币的共识规则，这将能增加比特币网络的整体安全性，以及减少个人用户被攻击。
+
+这些欺诈行为的证据可以被添加到证人的数据结构作为未来软叉的一部分，他们将帮助SPV客户强制规定，就连交易不使用segwit功能都可。
+
+## 不验证签名效率提高
+
+历史交易签名可能不比未来签名有趣 - 例如，比特币核心最近的检查点在默认情况下并没有检查交易签名，有的SPV客户干脆不检查数字签名本身，
+相信已经由矿工或其他节点检查。目前，签名数据事务处理的一个组成部分，必须存在，以便计算该事务散列。
+
+分离署名数据允许签名数据从磁盘进行清理，避免下载它，节约资源的节点。
+
+### 谁受惠?
+随着越来越多的事务使用segwit地址，人们或SPV节点将能够用更少的数据与磁盘空间进行操作。
+
+## 容量增加
+旧节点将仅供下载，只执行对这些数据的1 MB大小限制规则。新的节点，理解见证数据的完整，因此免费用新的来取代这限制，允许更大的数据。
+因此隔离见证利用这个机会提高限制为近4 MB，并增加了新的成本限制，以确保留在自己的资源利用平衡（这有效地限制接近1.6至2 MB）。
+
+### 谁受惠?
+
+谁运行升级钱包人们将能够通过移动签名交易的见证人部分利用增加的块大小。
+
+## 迈向单一组合限制
+
+目前，有两个保守的强制限制：该单一组合不超过 1MB，并在整个交易执行不可超过 20,000 签名检查。
+
+寻找最有利的交易集合给出单一的限制以包括是背包问题，可以用一个简单的算法来容易地解决几乎完美的实例。然而加入第二约束使得在某些情况下，
+很难找到一个很好的解决方案，这一理论问题在实践中已经迫使单一数据远低于容量待开采。
+
+如没有 hardfork 或基本上减小单一数据是无法解决这个问题。由于 segwit 无法修复，在不使情况变得更糟情况下：与其引入一个独立限制分离的见证数据，
+它采用单一限制适用于UTXO数据的加权和证人数据，同时允许合并后的实体限制。
+
+### 谁受惠?
+
+未来hardfork如能改变单一容量限制，使用者就能受惠。 如：
+
+    50* sigops + 4*basedata + 1*witnessdata < 10M
+
+使用者能轻易准确地填写单一数据同时最大限度并提高手续费收入，这让他们更可靠地计算出需要进行开采其交易的费用让用户受益。
+
+### 更多资料
+
+ * [背包问题](https://en.wikipedia.org/wiki/Knapsack_problem)
+ * [2015 年 8 月 Sigop 攻击 bitcoin 的讨论](https://bitcointalk.org/index.php?topic=1166928.0;all)
+ * [Gregory Maxwell 对比特币-dev证人的限制 ](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2015-December/011870.html)
```

Comment:
sorry , I don't speak manadarin , just paying for translation as no one else is translating these documents . Thus the confusion. Thanks for clarifying. Updated